server/src/main/java/org/apache/druid/server/coordination/SegmentLoadDropHandler.java

```diff
@@ -141,12 +143,15 @@ public Map<String, SegmentRowCountDistribution> getRowCountDistributionPerDataso
     return segmentManager.getRowCountDistribution();
   }
 
-  @Override
-  public void addSegment(DataSegment segment, @Nullable DataSegmentChangeCallback callback)
+  public void addSegment(
```

Comment:
We could have an internal overloaded method instead if we wanted to keep the interface

Reply:
Replied to the other comment.
No point adding two methods since only one of them is ever meant to be used in practice.

The contract of `SegmentLoadDropHandler` can probably be cleaned up further by making `addSegment` and `removeSegment` private. But that didn't seem necessary now.